### PR TITLE
Disable _snap kitchen-test on docker ubuntu

### PR DESCRIPTION
**Pull request by tpowell-progress**
## Description
We have a VM for Ubuntu that still tests this.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] If `Gemfile.lock` has changed, I have used `--conservative` to do it and included the full output in the Description above.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Review by jaymzh — APPROVED** [review]
Maybe add a comment explaining why?

File: kitchen-tests/cookbooks/end_to_end/recipes/linux.rb [hunk 1]
```diff
@@ -217,4 +217,4 @@
 #   end
 # end
 
-include_recipe "::_snap" if platform?("ubuntu")
+include_recipe "::_snap" if platform?("ubuntu") && !File.exist?("/.dockerenv")
```
Comment (Stromweld) [inline kitchen-tests/cookbooks/end_to_end/recipes/linux.rb:220]:
```suggestion
include_recipe "::_snap" if platform?("ubuntu") && !docker?
```

**Comment by jaymzh** [comment]
I don't think this is actually the answer.

Because the VM is getting the exact same situation:

```
        [2024-11-21T17:06:26+00:00] FATAL: Chef::Exceptions::Package: snap_package[hello] (end_to_end::_snap line 11) had an error: Chef::Exceptions::Package: {"message"=>"Get \"https://api.snapcraft.io/v2/snaps/info/hello?architecture=amd64&fields=architectures%2Cbase%2Cconfinement%2Clinks%2Ccontact%2Ccreated-at%2Cdescription%2Cdownload%2Cepoch%2Clicense%2Cname%2Cprices%2Cprivate%2Cpublisher%2Crevision%2Csnap-id%2Csummary%2Ctitle%2Ctype%2Cversion%2Cwebsite%2Cstore-url%2Cresources%2Cmedia%2Ccommon-ids%2Ccategories\": context canceled"}
```

**Comment by jaymzh** [comment]
I updated this to address code review comments, because I was going to merge it but given the same errors here and on VMs, it seems prudent to wait.

**Comment by tpowell-progress** [comment]
> I updated this to address code review comments, because I was going to merge it but given the same errors here and on VMs, it seems prudent to wait.

Yeah, I could've sworn that Ubuntu VMs were in good shape but I see them failing too.

**Pull request closed without merging**